templates/tfengine/components/cicd/variables.tf

```diff
@@ -12,6 +12,28 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License. */ -}}
 {{$props := .__schema__.properties -}}
+{{$csrProps := $props.cloud_source_repository.properties -}}
+{{$githubProps := $props.github.properties -}}
+{{$envsProps := $props.envs.items.properties -}}
+{{$triggerProps := $envsProps.triggers.properties -}}
+variable "parent_id" {
+  type        = string
+  description = {{schemaDescription $props.parent_id.description}}
+  validation {
+    condition     = can(regex("{{$props.parent_id.pattern}}", var.parent_id))
+    error_message = "The parent_id must be valid. Should have only numeric values with a length between 8 and 25 digits. See https://cloud.google.com/resource-manager/docs/cloud-platform-resource-hierarchy to know how to get your organization/folder id."
+  }
+}
+
+variable "parent_type" {
+  type        = string
+  description = {{schemaDescription $props.parent_type.description}}
+  validation {
+    condition     = can(regex("{{$props.parent_type.pattern}}", var.parent_type))
```

Comment:
Should use the terraformPattern function here and in parent_id

Reply:
You're right. I'm sorry. Started with this one first and forgot it 😅